components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-list.jsp

```diff
@@ -30,166 +50,267 @@
-                                       onclick="editIdPName('<%=Encode.forJavaScriptAttribute(identityProvidersList.get(i).getIdentityProviderName())%>');return false;"
-                                       href="#"
-                                       class="icon-link"
-                                       style="background-image: url(images/edit.gif)">
+                                       onclick="editIdPName('<%=Encode.forJavaScriptAttribute(idp.getIdentityProviderName())%>');return false;"
+                                       style="background-image: url(images/edit.gif);" class="icon-link">
                                        <fmt:message key='edit'/>
                                     </a>
                                     <a title="<fmt:message key='delete'/>"
-                                       onclick="deleteIdPName('<%=Encode.forJavaScriptAttribute(identityProvidersList.get(i).getIdentityProviderName())%>');return false;"
+                                       onclick="deleteIdPName('<%=Encode.forJavaScriptAttribute(idp.getIdentityProviderName())%>', '<%=pageNumberInt%>', '<%=Encode.forJavaScriptAttribute(filterString)%>');return false;"
                                        href="#"
                                        class="icon-link"
                                        style="background-image: url(images/delete.gif)">
                                        <fmt:message key='delete'/>
                                     </a>
-                                </td>
-                            </tr>
-                        <% } %>
-                    <% } else { %>
-                        <tr>
-                            <td colspan="3"><i><fmt:message key='no.idp'/></i></td>
-                        </tr>
-                    <% } %>
-                </tbody>
+                                 </td>
+                              </tr>
+                              <%
+                                 }
+                                 %>
+                           </tbody>
+                           <% } else { %>
+                           <tbody>
+                              <tr>
+                                 <td colspan="3"><i>No Service Providers registered</i></td>
+                              </tr>
+                           </tbody>
+                           <% } %>
+                        </table>
+                     </td>
+                  </tr>
+               </tbody>
             </table>
-            </div>
-        </div>
-    </div>
-
+            <br/>
+            <carbon:paginator pageNumber="<%=pageNumberInt%>"
+               numberOfPages="<%=numberOfPages%>"
+               page="idp-mgt-list.jsp"
+               pageNumberParameterName="pageNumber"
+               resourceBundle="org.wso2.carbon.idp.mgt.ui.i18n.Resources"
+               parameters="<%=paginationValue%>"
```

Comment:
encode this since paginationValue contains filterString which is a request parameter